evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -227,20 +227,21 @@ def test_additional_objectives(X_y_binary):
 
 
 @patch('evalml.objectives.BinaryClassificationObjective.optimize_threshold')
+@patch('evalml.pipelines.BinaryClassificationPipeline._encode_targets', side_effect=lambda y: y)
```

Comment:
need to patch `encode_targets` in order to optimize the threshold

Reply:
Yep, otherwise sklearn will throw an exception. The encoder is fit during `pipeline.fit` but since we're mocking fit, that doesn't happen.